src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java

```diff
@@ -1255,7 +1255,7 @@ public <C extends Credentials> List<C> getCredentials(@NonNull Class<C> type,
                                                           @NonNull Item item,
                                                           @Nullable org.acegisecurity.Authentication authentication) {
         Objects.requireNonNull(item);
-        return getCredentialsInItemGroup(type, item.getParent(), authentication == null ? null : authentication.toSpring(), List.of());
+        return getCredentialsInItemFallback(type, item, authentication == null ? null : authentication.toSpring(), List.of());
```

Comment:
Prevents infinite recursion in case the overrides delegates to super implementation.